src/domain/valueObjects/Event.ts

```diff
@@ -8,45 +8,65 @@ import { Parser } from '../../interfaces/Parser';
  * @description Create the main type of event, the titular `Event` which
  * can be reassembled to other, more specific types or "value objects" later.
  */
-export function makeEvent(
+export async function makeEvent(
   parser: Parser,
   body: Record<string, any>,
   headers: Record<string, any>
-): Event {
+): Promise<Event> {
   const eventConcrete = new EventConcrete(parser, body, headers);
+  await EventConcrete.populate(eventConcrete, parser, body, headers);
   return eventConcrete.getDTO();
 }
 
 class EventConcrete {
-  repo: string;
+  repo: string = ''
   date: string;
-  eventType: string;
-  id: string;
-  changes: Change[];
-  eventTime: string;
-  timeCreated: string;
-  timeResolved: string;
-  title: string;
-  message: string;
+  eventType: string = ''
+  id: string = '';
+  changes: Change[] = new Array<Change>(0);
```

Comment:
Seems like we can simplify this by having it simply as `changes: Change[] = [];`.